src/praisonai/praisonai/integrations/compute_managed_agent.py

```
@@ -232,6 +247,25 @@ async def _write(self, path: str, content: str) -> None:
+    """Reclaim an instance whose backend is gone.
+
+    Registered with weakref.finalize, so it runs when the backend is collected
+    *and* at interpreter exit. It must not close over the backend itself --
+    that would keep it alive and prevent the very collection it waits for.
+    """
+    import asyncio
+
+    try:
+        asyncio.run(provider.shutdown(instance_id))
+        logger.debug("[compute_managed] released %s on %s", instance_id, place)
+    except Exception as exc:  # pragma: no cover - teardown stays quiet
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)